python/yugabyte_db_thirdparty/yb_build_thirdparty_main.py

```diff
@@ -93,6 +94,23 @@ def main() -> None:
         log("Time taken for packaging/upload %.1f sec",
             time.time() - packaging_and_upload_start_time_sec)
 
+    if builder.args.snyk:
+        snyk_token = os.environ.get('SNYK_TOKEN')
+        if snyk_token is None:
+            log("SNYK_TOKEN is not set, not running snyk.")
+        else:
+            log("Running Snyk Vulnerability Scan.")
+            os_type = os.environ.get('OSTYPE', '')
+            if os_type.startswith('linux'):
+                os.system("curl https://static.snyk.io/cli/latest/snyk-linux -o snyk")
+                os.system("chmod +x ./snyk")
+                rc = os.system(f"./snyk auth {snyk_token}")
+                if rc != 0:
+                    log("Snyk authentication failed. Aborting scan.")
+                else:
+                    homedir = os.environ.get('YB_THIRDPARTY_DIR')
+                    os.system(f"./snyk monitor {homedir}/src --unmanaged")
+
```

Comment:
If this fails, there should be a nonzero exit code from the build script. `subprocess.check_call` takes care of this.